iocore/hostdb/HostDB.cc

```diff
@@ -625,29 +640,34 @@ probe(const Ptr<ProxyMutex> &mutex, HostDBHash const &hash, bool ignore_timeout)
   }
 
   // Otherwise HostDB is enabled, so we'll do our thing
-  ink_assert(this_ethread() == hostDB.refcountcache->lock_for_key(hash.hash.fold())->thread_holding);
-  uint64_t folded_hash = hash.hash.fold();
+  uint64_t folded_hash           = hash.hash.fold();
+  std::shared_mutex &bucket_lock = hostDB.refcountcache->lock_for_key(folded_hash);
```

Comment:
Why not use ts::shared_mutex to avoid the risk of writer starvation?

Reply:
Switched to ts::shared_mutex.